SectionedSolidHorizontal-HorizontalLinearDirectrix-1/UT-SectionedSolidHorizontal-LinearDirectrix-HorizontalElement-1.ifc

```diff
@@ -0,0 +1,296 @@
+ISO-10303-21;
+HEADER;FILE_DESCRIPTION(('ViewDefinition [Ifc4X3NotAssigned]'),'2;1');
+FILE_NAME('BridgeHorizontalIShape.ifc',
+'2021-08-05T11:08:35',('Nadia Sugar'),('No Organization',''),
+'EDMsix Version 2.0.209.92 Dec 12 2020',
+'Allplan 99.0 04.08.2021 - 22:13:56','');
+FILE_SCHEMA(('IFC4X3_RC4'));
+ENDSEC;
+
+DATA;
+#1= IFCPERSON($,'Sugar','Nadia',$,$,$,$,$);
+#3= IFCORGANIZATION($,'No Organization',$,$,$);
+#7= IFCPERSONANDORGANIZATION(#1,#3,$);
+#10= IFCAPPLICATION(#3,'99.0','Allplan','Allplan');
+#11= IFCOWNERHISTORY(#7,#10,$,.NOTDEFINED.,$,$,$,1628152120);
+#12= IFCSIUNIT(*,.LENGTHUNIT.,$,.METRE.);
+#13= IFCSIUNIT(*,.LENGTHUNIT.,.DECI.,.METRE.);
+#14= IFCSIUNIT(*,.AREAUNIT.,.DECI.,.SQUARE_METRE.);
+#15= IFCSIUNIT(*,.VOLUMEUNIT.,.DECI.,.CUBIC_METRE.);
+#16= IFCSIUNIT(*,.MASSUNIT.,.KILO.,.GRAM.);
+#17= IFCSIUNIT(*,.PLANEANGLEUNIT.,$,.RADIAN.);
+#18= IFCMEASUREWITHUNIT(IFCPLANEANGLEMEASURE(0.0174532925199433),#17);
+#19= IFCDIMENSIONALEXPONENTS(0,0,0,0,0,0,0);
+#20= IFCCONVERSIONBASEDUNIT(#19,.PLANEANGLEUNIT.,'DEGREE',#18);
+#22= IFCAXIS2PLACEMENT3D(#25,$,$);
+#25= IFCCARTESIANPOINT((0.,0.,0.));
+#27= IFCPROJECT('1VWTA2AQn4bezaJ7736V_D',#11,'Bridge 5',$,$,$,$,(#60),
+#37);
+#37= IFCUNITASSIGNMENT((#13,#14,#15,#16,#20));
+#39= IFCBRIDGE('2qX8Zx6Lv4oxQKYStwzyUq',#11,'Bridge5',$,'',#53,$,$,
+.ELEMENT.,.USERDEFINED.);
+#50= IFCAXIS2PLACEMENT3D(#51,$,$);
+#51= IFCCARTESIANPOINT((0.,0.,0.));
+#53= IFCLOCALPLACEMENT($,#50);
+#56= IFCGEOMETRICREPRESENTATIONSUBCONTEXT('Body','Model',*,*,*,*,#60,$,
+.MODEL_VIEW.,$);
+#60= IFCGEOMETRICREPRESENTATIONCONTEXT($,'Model',3,1.E-5,#22,$);
+#63= IFCAXIS2PLACEMENT3D(#64,$,$);
+#64= IFCCARTESIANPOINT((1.,-35.65,-0.693));
+#66= IFCLOCALPLACEMENT(#53,#63);
+#67= IFCPRODUCTDEFINITIONSHAPE('Explicit Shape',$,(#467));
+#71= IFCPOLYLINE((#73,#75));
+#73= IFCCARTESIANPOINT((0.,0.,0.));
+#75= IFCCARTESIANPOINT((431.,0.,0.));
+#77= IFCSECTIONEDSOLIDHORIZONTAL(#71,(#79,#263),(#446,#452),.F.);
+#79= IFCARBITRARYCLOSEDPROFILEDEF(.AREA.,'',#82);
+#82= IFCCOMPOSITECURVE((#90,#99,#107,#115,#126,#134,#145,#153,#164,#172,
+#180,#188,#196,#204,#215,#223,#234,#242,#253,#261),$);
+#84= IFCPOLYLINE((#86,#88,#86));
```

Comment:
The polyline segments within your composite profile curve return on themselves.  Ie #86 appears twice on the line above.  This doesn't seem right.  Did you also consider using an IfcIndexedPolyCurve to define the profile? It is a simpler an more efficient way to do so.  https://standards.buildingsmart.org/IFC/RELEASE/IFC4/ADD2_TC1/HTML/link/ifcindexedpolycurve.htm